util/fibers/detail/fiber_interface.cc

```diff
@@ -275,14 +275,23 @@ void FiberInterface::PrintStackTrace() {
     state = "active";
   }
 
-  auto print = [this, state]() {
-    LOG(INFO) << "------------ Fiber " << this->name_ << " (" << state << ") ------------\n" << GetStacktrace();
+  auto print_fn = [state](FiberInterface* fb) {
+    LOG(INFO) << "------------ Fiber " << fb->name_ << " (" << state << ") ------------\n"
+              << GetStacktrace();
   };
 
+  PrintStackTrace(std::move(print_fn));
+}
+
+void FiberInterface::PrintStackTrace(PrintFn fn) {
```

Comment:
Let's rename this into `ExecuteInFiber` or something :)

Reply:
ok, by the way, do you understand what lines 295-300 do? if yes, I would like to add comments there, @royjacobson .